src/pages/homePage/HomePage.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react';
+import { NavLink } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchDetails, showModal, updateCurrency } from '../../redux/features/detailsSlice';
+import CurrencyHome from '../../components/currencyCard/CurrencyCard';
+import Navbar from '../../components/navbar/Navbar';
+import './HomePage.css';
+
+const HomePage = () => {
+  const dispatch = useDispatch();
+  const { currencyList, status, modal } = useSelector((store) => store.details);
+  const [type, setType] = useState('usd');
+
+  const handleModal = () => {
+    dispatch(showModal());
+  };
+
+  const handleType = (name) => {
+    setType(name);
+    dispatch(updateCurrency(name));
+  };
+
+  useEffect(() => {
+    if (status === 'idle') {
+      dispatch(fetchDetails(type));
+    }
+  }, [status, dispatch, type]);
+
+  if (status === 'loading') return <div>loading...</div>;
+
+  return (
+    <div className="home-page">
+      <Navbar type="home" />
+      {modal && (
+        <div className="modal">
+          <p onClick={() => {
+            handleType('usd');
+            handleModal();
+          }}
+          >
+            Usd
+          </p>
+          <p onClick={() => {
+            handleType('eur');
+            handleModal();
+          }}
+          >
+            Euro
+          </p>
+          <p onClick={() => {
+            handleType('gbp');
+            handleModal();
+          }}
+          >
+            Pound
+          </p>
+          <p onClick={() => {
+            handleType('jpy');
+            handleModal();
+          }}
+          >
+            Japanese Yen
+          </p>
+          <p onClick={() => {
+            handleType('cad');
+            handleModal();
+          }}
+          >
+            Canadian Dollar
+          </p>
+        </div>
+      )}
+      <div className="details-home-container">
```

Comment:
- You have done well trying to make changes suggested by the previous reviewer, but you have not exactly added the filter feature. It should be possible for users to filter the list of cryptocurrencies on your home page, either by the coin name or by price. I suggest adding an input form that could be of help if you decide to filter by coin name. Please fix this 🙏

Reply:
I dont understand you. I am filtering by currency. If you click on the different currency it will bring based on the crypto based on the currency that you want.

Reply:
Okay. I have now implemented the changes. I filter using price range, and the design is for both mobile and desktop

Reply:
@Fene-87, the reviewer meant that you should have for example a search bar where the user can search for a particular coin on the page. this will be implemented using the filter funtion.